system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj

```diff
@@ -0,0 +1,180 @@
+  [(ingest/reset-fixture {"provguid1" "PROV1"})
+   service/grant-all-service-fixture]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Helper utility functions
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn- get-svc
+  "Ingest a single svc given a svc-name."
+  [svc-name]
+  (service/ingest-service-with-attrs {:Name svc-name}))
+
+(defn- get-updated-svc
+  "Update (re-ingest) an existing (old) service with new data."
+  [old-svc data]
+  (index/wait-until-indexed)
+  (service/ingest-service-with-attrs
+   (merge data
+          {:native-id (:native-id old-svc)})))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Retrieve by concept-id - general test
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(deftest retrieve-service-by-concept-id-any-accept-header
+  (let [svc1-name "service1"
+        svc1-name-new "service1-new"
+        del-svc-name "deleted-service"
+        svc1-v1 (get-svc svc1-name)
+        del-svc (get-svc del-svc-name)
+        _ (index/wait-until-indexed)
+        del-concept (mdb/get-concept (:concept-id del-svc))]
+    (ingest/delete-concept del-concept
+                           (service/token-opts (e/login (s/context) "user1")))
+    (index/wait-until-indexed)
+    (testing "retrieval of a deleted service results in a 404"
+      (let [{:keys [status errors]} (search/get-search-failure-xml-data
+                                     (search/retrieve-concept
+                                      (:concept-id del-svc)
+                                      nil
+                                      {:accept mt/any
+                                       :throw-exceptions true}))]
+        (is (= 404 status))
+        (is (= [(format "Concept with concept-id [%s] could not be found."
+                        (:concept-id del-svc))]
+               errors))))
+    (let [svc1-v2 (get-updated-svc svc1-v1 {:Name svc1-name-new})]
+      (testing (str "Sanity check that the test service got updated and its "
+                    "revision id was incremented.")
+        (is (= 2
+               (inc (:revision-id svc1-v1))
+               (:revision-id svc1-v2))))
+      (let [response (search/retrieve-concept
+                      (:concept-id svc1-v1)
+                      nil
+                      {:accept mt/any})
+            response-v1 (search/retrieve-concept
+                         (:concept-id svc1-v1)
+                         1
+                         {:accept mt/any})
+            response-v2 (search/retrieve-concept
+                         (:concept-id svc1-v1)
+                         2
+                         {:accept mt/any})]
+          (testing "retrieval by service concept-id returns the latest revision."
+            (is (= svc1-name-new
```

Comment:
How about creating a helper function (e.g. `verify-retrieved-concept`) to do the whole retrieving concept, parsing JSON and comparing service name in just one function?